fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -156,14 +162,19 @@ def loop_over_benchmark_problems(problem_group, options, checkpointer):
                 loop_over_starting_values(problem,
                                           options=options,
                                           grabbed_output=grabbed_output,
-                                          checkpointer=checkpointer)
+                                          checkpointer=checkpointer,
+                                          emissions_tracker=emissions_tracker)
             results.extend(problem_results)
             failed_problems.extend(problem_fails)
 
+            if emissions_tracker:
+                _ = emissions_tracker.stop()
```

Comment:
Is this only needed after a `start()`?
We can probably run it after the `try` block, where we clear runtime and `clean_matlab` on ~ln660.

Reply:
I followed the [example in the codecarbon docs](https://mlco2.github.io/codecarbon/usage.html#explicit-object) which runs a final `stop()` after the emissions for each task have been tracked. I think moving this to where runtime/matlab are cleared would mean that a new `EmissionsTracker` object would have to be created for the next fit?

Reply:
Ahh yeah, I was thinking of `task_stop()`. So this calls the final `stop()` after each problem?
Not sure if you need to call `stop_task()` before you can `start_task()` again but if there's an error in the try block, then the task won't stop.

Reply:
Ah yeah that's a good point - I'll take another look at this

Reply:
I've added an extra line after the calls to `clear_matlab` and clearing runtime which stops the emissions tracker task if the `emissions_tracker` has been created but the value for `emissions` hasn't been set